manualsource/src/main/java/org/hibernate/search/manualsource/impl/FullTextQueryImpl.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.search.manualsource.impl;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.lucene.search.Explanation;
+import org.apache.lucene.search.Filter;
+import org.apache.lucene.search.Sort;
+
+import org.hibernate.search.engine.integration.impl.ExtendedSearchIntegrator;
+import org.hibernate.search.exception.AssertionFailure;
+import org.hibernate.search.filter.FullTextFilter;
+import org.hibernate.search.manualsource.FullTextQuery;
+import org.hibernate.search.manualsource.source.ObjectInitializer;
+import org.hibernate.search.query.engine.QueryTimeoutException;
+import org.hibernate.search.query.engine.spi.EntityInfo;
+import org.hibernate.search.query.engine.spi.FacetManager;
+import org.hibernate.search.query.engine.spi.HSQuery;
+import org.hibernate.search.query.engine.spi.TimeoutExceptionFactory;
+import org.hibernate.search.query.engine.spi.TimeoutManager;
+import org.hibernate.search.spatial.Coordinates;
+import org.hibernate.search.spatial.impl.Point;
+import org.hibernate.search.util.logging.impl.Log;
+import org.hibernate.search.util.logging.impl.LoggerFactory;
+
+/**
+ * @author Emmanuel Bernard &lt;emmanuel@hibernate.org&gt;
+ */
+public class FullTextQueryImpl implements FullTextQuery {
+
+	private static final Log log = LoggerFactory.make();
+	private final WorkLoadImpl workLoadImpl;
+
+	private final HSQuery hSearchQuery;
+
+	/**
+	 * Constructs a  <code>FullTextQueryImpl</code> instance.
+	 */
+	public FullTextQueryImpl(org.apache.lucene.search.Query query,
```

Comment:
Ok, so this would not work with ES. I suppose we are fine on the indexing side, but querying would need some enabling work.